pkgs/tools/nix/nixpkgs-fmt/default.nix

```diff
@@ -1,16 +1,16 @@
 { lib, rustPlatform, fetchFromGitHub, fetchpatch }:
 rustPlatform.buildRustPackage rec {
   pname = "nixpkgs-fmt";
-  version = "0.9.0";
+  version = "1.0.0";
 
   src = fetchFromGitHub {
     owner = "nix-community";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1kkw87c63nx5pqsxcwn6iw27k02j9ls21zyhb5dvf0zaqd9sz7ad";
+    sha256 = "sha256-rO30/B+mH/ZOEH8IF2fF4uHK7XQqyR4zIDueHnmNMHA=";
   };
 
-  cargoSha256 = "1wybvm9qckx9cd656gx9zrbszmaj66ihh2kk6qqdb6maixcq5k0x";
+  cargoSha256 = "sha256-fP/8yDg+W7hiWGucbRr338y9PEDTb2bdq71JKE6M8OM=";
```

Comment:
I believe from reading https://github.com/NixOS/nixpkgs/pull/89308 that sri and base64 hashes are fine, it just that the fetcher function needs to support it, and fetchgit doesn't yet https://github.com/NixOS/nixpkgs/pull/79987. (though this is going to use fetchzip/fetchurl under the hood which does). So can you please use `hash` instead of `sha256`?

Though for `cargoSha256` there isn't a `cargoHash`, so I think it would make sense for this to just be base64 encoded, but not sri.

Reply:
the user may need to rollback their version of `nix`, as I'm not aware of a way to configure the hash used by fetchers

Reply:
You just have to do
```
 nix to-base64 "sha256-rO30/B+mH/ZOEH8IF2fF4uHK7XQqyR4zIDueHnmNMHA="
```

to make it non-sri. 

I didn't mean you have to configure the fetcher, it was just
```diff
fetchFromGitHub = {
-sha256 = ...
+hash = ...
```

Reply:
I was talking about the case where nix kicks back at you and says:
```
  wanted: ...
  actual: .. 
```
I think in nix 2.4, it will automatically make this base64

Reply:
@worldofpeace 

I got `BadHash` error when running `nix-build -A nixpkgs-fmt` inside nixpkgs:
```
error: --- BadHash ------------------------------------------------------------------------------------------- nix-build
hash 'rO30/B+mH/ZOEH8IF2fF4uHK7XQqyR4zIDueHnmNMHA=' does not include a type
(use '--show-trace' to show detailed location information)
```

`nix --version` = `nix (Nix) 2.4pre20200721_ff314f1`

this is the code:
```
  src = fetchFromGitHub {
    owner = "nix-community";
    repo = pname;
    rev = "v${version}";
    hash = "rO30/B+mH/ZOEH8IF2fF4uHK7XQqyR4zIDueHnmNMHA=";
  };

  cargoSha256 = "fP/8yDg+W7hiWGucbRr338y9PEDTb2bdq71JKE6M8OM=";
```